internal/control_loop/direct.go

```diff
@@ -52,10 +52,8 @@ func (l *DirectControlLoop) Cycle(target int) int {
 
 	l.lastOutput = stepTarget
 
-	// convert the result to an integer
-	rounded := int(math.Round(stepTarget))
 	// ensure we are within sane bounds
-	result := util.Coerce(rounded, 0, 255)
+	result := util.Coerce(stepTarget, 0, 255)
```

Comment:
I would have thought that changing the implementation to use percentages would also change this clamping to `[0, 100]`, is this intentionally left at `255`?

Reply:
Look more closely at the code: I didn't change the implementation to use percentages.
Internally it uses exactly the same as before, those values between 0 and 255 that are scaled to MinPwm..MaxPwm.

But if a step has the value `42%` (instead of just `42`), that is converted to a value between 0 and 255 (by multiplying it with 255/100 = 2.55).

This minimizes potential for internal breakage and ensures backwards-compatibility